modules/core/src/main/java/org/apache/ignite/internal/properties/IgniteProductVersion.java

```diff
@@ -47,11 +48,15 @@ public class IgniteProductVersion implements Serializable {
     /** Flag indicating if this is a snapshot release. */
     private final boolean isSnapshot;
 
-    private IgniteProductVersion(byte major, byte minor, byte maintenance, boolean isSnapshot) {
+    /** Alpha version part or an empty string if this is not an alpha release. */
+    private final String alphaVersion;
```

Comment:
Why not byte?

Reply:
Because I don't know if we actually need to change it a little bit in order to accommodate beta version, for example. We can switch to a byte later if needed.

Reply:
Ok fix it in https://issues.apache.org/jira/browse/IGNITE-17146